tools/esdlcomp/esdlcomp.h

```diff
@@ -621,6 +621,45 @@ class ParamInfo
         out.append("/>\n");
     };
 
+    bool checkDup(StringBuffer& msg, ParamInfo* attrlist)
+    {
+        if(!attrlist)
+            return false;
+        StringBuffer myeclnamebuf;
+        bool hasMyEclName = getMetaStringValue(myeclnamebuf, "ecl_name");
+        ParamInfo* attr = attrlist;
+        while(attr)
+        {
+            if(stricmp(name, attr->name) == 0)
```

Comment:
You could really simplify by commoning up these 4 similar compare / message building steps into inline function(s).  But this works.